modules/internal/CString.chpl

```diff
@@ -116,11 +116,18 @@ module CString {
   //
   // casts from c_string to bool types
   //
-  inline proc _cast(type t, x:c_string) where isBoolType(t)
+  inline proc _cast(type t, x:c_string) throws where isBoolType(t)
   {
     pragma "insert line file info"
-    extern proc c_string_to_chpl_bool(x:c_string) : bool;
-    return c_string_to_chpl_bool((x:string).strip().c_str()) : t;
+    extern proc c_string_to_chpl_bool(x:c_string, ref err: bool) : bool;
+
+    var isErr: bool;
+    var retVal = c_string_to_chpl_bool((x:string).strip().c_str(), isErr): t;
+
+    if isErr then
+      throw new IllegalArgumentError(x:string, "Unexpected value when converting from string to bool");
+
+    return retVal;
```

Comment:
Why isn't this method written the same as the string version?
Why aren't the methods below updated (e.g. `inline proc _cast(type t, x:c_string) where t == int(8)
return c_string_to_int8_t((x:string).strip().c_str());`)?
Since these c_string casts call strip anyway, can they just convert the c_string to a string and then use the string : int (or whatever) cast?